utils/crypto/transaction.js

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable security/detect-object-injection */
+/**
+ * Check if balance is available to make successful transaction
+ * @param {Object} debit - amount to be deduced (coins object)
+ * @param {Object} available - actual amount avaliable (conins object)
+ */
+function checkSufficientAmountAvaliable (debit, available) {
+  const currenciesAvailableStatus = []
+  for (const key in debit) {
+    if (Object.hasOwnProperty.call(debit, key) && Object.hasOwnProperty.call(available, key)) {
+      const debitCurrency = debit[key]
+      const availableCurrency = available[key]
+      if (debitCurrency <= availableCurrency) {
+        currenciesAvailableStatus.push(true)
+      } else {
+        currenciesAvailableStatus.push(false)
+      }
+    }
+  }
+  const status = !(currenciesAvailableStatus.some((val) => val === false))
```

Comment:
Can we use simple Boolean filter here

Reply:
Or we can have a key named  someTrue or someFalse rather than doing traversal of whole array